LetsDoStuff.WebApi/Controllers/SubscribingController.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using LetsDoStuff.WebApi.Services;
+using LetsDoStuff.WebApi.Services.DTO;
+using LetsDoStuff.WebApi.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LetsDoStuff.WebApi.Controllers
+{
+    [Route("api/subscribing")]
+    public class SubscribingController : ControllerBase
+    {
+        private readonly ISubscribingService _subscribingService;
+
+        public SubscribingController(ISubscribingService subscribingService)
+        {
+            _subscribingService = subscribingService;
+        }
+
+        [HttpGet("SubscribeToActivityById")]
+        [Authorize]
+        public IActionResult SubscribeToActivityById(int idActivity)
+        {
+            var loginName = this.HttpContext.User.Claims.First().Value;
+            try
+            {
+                _subscribingService.MakeUserSubscribedToActivityByEmailAndId(loginName, idActivity);
+                return Ok();
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpGet("UnsubscribeToActivityById")]
+        [Authorize]
+        public IActionResult UnsubscribeToActivityById(int idActivity)
+        {
+            var loginName = this.HttpContext.User.Claims.First().Value;
+            try
+            {
+                _subscribingService.MakeUserUnsubscribedToActivityByEmailAndId(loginName, idActivity);
+                return Ok();
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpGet("subscriberinfo")]
+        [Authorize]
+        public ActionResult<List<ActivityResponse>> GetSubscriberInfo()
+        {
+            var userName = this.HttpContext.User.Claims.FirstOrDefault().Value;
+            var subscriberinfo = _subscribingService.GetAllActivitiesOfTheUser(userName);
```

Comment:
If It was possible to use it without changing authorization I'd agree with you. But here I use userName ( needed to rename user email or just email ) because I could find it by authorization's files. But ok, I'll create another method for searching by id.

Reply:
Can't we just save id in 'authorization's files'? =)

Reply:
> Can't we just save id in 'authorization's files'? =)

I'm not sure about this. You see, we use Claim and ClaimsIdentity in AccountController for authorization, so in ClaimsIdentity we have only two fields for creation a Claim: DefaultNameClaimType and DefaultRoleClaimType. I could remove the email and put the id instead but I'm afraid it would be terrible because many of us already use  this structure to identify a user ( for instance, Roma's used "HttpContext.User.Identity.Name" for returning an email in GetUserSettings )
I could try just to add this thing like an extra parameter but  I'm not sure would it work in that case.  So I think it would be better to do this in another task and we need to be very careful doing this.

Reply:
>we have only two fields for creation a Claim
the mistake is here =) please take a better look at the Claim constructor you used in the auth controller, does it really restrict number or names of the claim?

Reply:
`// ...`
`var claims = new List<Claim>
                      {
                        new Claim("Id", user.Id.ToString()),
                        new Claim(ClaimsIdentity.DefaultNameClaimType, user.Email),
                        new Claim(ClaimsIdentity.DefaultRoleClaimType, user.Role)
                    };`
`// ...`
`var response = new
            {
                access_token = encodedJwt,
                login = identity.Name,
                id = identity.FindFirst("Id").Value
            };`